src/main/java/model/producer/ExcavatorProducer.java

```diff
@@ -0,0 +1,20 @@
+package model.producer;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer {
```

Comment:
Do not use raw types. Always parametrize interfaces and classes if it is possible. In this case, you should ensure that `MachineProducer` cannot be implemented with a type that is not a subtype of `Machine`. To fix this, you should define the `MachineProducer` interface with a generic type that extends `Machine`.